src/Generating/FinishGen.cpp

```diff
@@ -124,9 +124,9 @@ void cFinishGenNetherClumpFoliage::TryPlaceClump(cChunkDesc & a_ChunkDesc, int a
 	}
 
 	int MaxY = a_RelY + 2;
-	if (MaxY > cChunkDef::Height)  // Check if the coordinate is outside the chunk. If it it then adjust it.
+	if (MaxY >= cChunkDef::Height)  // Check if the coordinate is outside the chunk. If it it then adjust it.
 	{
-		MaxY = cChunkDef::Height;
+		MaxY = cChunkDef::Height - 1;
```

Comment:
The original code was correct, `MaxY` is used as an non-inclusive bound in the for loop below.